test/parallel/test-buffer.js

```diff
@@ -1163,3 +1163,34 @@ assert.throws(function() {
   var b = new Buffer(1);
   b.equals('abc');
 });
+
+// Test indexOf
+var b = new Buffer('abcdef');
+
+assert.equal(b.indexOf('a'), 0);
+assert.equal(b.indexOf('a', 1), -1);
+assert.equal(b.indexOf('a', -1), 0);
+assert.equal(b.indexOf('a', NaN), 0);
+assert.equal(b.indexOf('a', -Infinity), 0);
+assert.equal(b.indexOf('a', Infinity), -1);
+assert.equal(b.indexOf('bc'), 1);
+assert.equal(b.indexOf('bc', 2), -1);
+assert.equal(b.indexOf('bc', -1), 1);
+assert.equal(b.indexOf('bc', NaN), 1);
+assert.equal(b.indexOf('bc', -Infinity), 1);
+assert.equal(b.indexOf('bc', Infinity), -1);
+assert.equal(b.indexOf('z'), -1);
+assert.equal(b.indexOf(''), 0);
+assert.equal(b.indexOf('', 1), 1);
+assert.equal(b.indexOf('', b.length + 1), b.length);
+assert.equal(b.indexOf('', Infinity), b.length);
```

Comment:
And these four `b.indexOf('')` calls would return `-1` because `Array.prototype.indexOf`/`TypedArray.prototype.indexOf` do a `===` check and no byte will ever be `byte === ''`